src/MapViewController.swift

```diff
@@ -128,6 +141,49 @@ public class MapViewController: TGMapViewController, LocationManagerDelegate, TG
     }
   }
 
+  public func display(route: OTRRoutingResult) throws {
+    //TODO: We eventually should support N number of routes.
+    if let routeMarker = currentRouteMarker {
+      //We don't throw if the remove fails here because we want to silently replace the route
+      currentRouteMarker = nil
+      markerRemove(routeMarker)
+    }
+    let routeLeg = route.legs[0]
+    let polyLine = TGGeoPolyline(size: UInt32(routeLeg.coordinateCount))
+
+    //TODO: Need to investigate more if this is a bug in OTR or if valhalla returns null island at the end of their requests?
```

Comment:
Yeah that's what I'm figuring. Here's the decode implementation:
https://github.com/mapzen/on-the-road_ios/blob/master/OTRRouting/OTRRoutingResultLeg.m#L53